daffodil-sapi/src/test/scala/org/apache/daffodil/example/TestScalaAPI.scala

```diff
@@ -945,4 +946,55 @@ class TestScalaAPI {
     assertEquals("over", outputter.getResult.text)
   }
 
+  /***
+   * This test uses the same input file as the CLI blob test, test_blob_backtracking.
+   * The necessary file may need to be generated, see instructions in:
+   *
+   * daffodil-cli/src/it/scala/org/apache/daffodil/blob/TestBlob.scala
+   *
+   * If it has already been generated, it can be copied from:
+   *
+   * daffodil/daffodil-cli/src/it/resources/org/apache/daffodil/CLI/input/2049MB.bin
+   *
+   * to:
+   *
+   * daffodil/daffodil-sapi/src/test/resources/test/sapi
+   *
+   ***/
+  /*@Test*/
+  def testScalaAPI19() {
+    // Demonstrate that we cannot use the API to continue a parse with an invalid InputSource
+    // ie. after a runtime SDE. This test needs to be run with an input file larger than 256MB
+    val c = Daffodil.compiler()
+    c.setValidateDFDLSchemas(false)
+    val schemaFile = getResource("/test/sapi/blob_backtracking.dfdl.xsd")
+    c.setDistinguishedRootNode("root", null)
+    val pf = c.compileFile(schemaFile)
+    val dp1 = pf.onPath("/")
+    val dp = reserializeDataProcessor(dp1)
+
+    val file = getResource("/test/sapi/2049MB.bin")
+    val fis = new java.io.FileInputStream(file)
+    val input = new InputSourceDataInputStream(fis)
+
+    val outputter = new ScalaXMLInfosetOutputter()
+    var res: ParseResult = null
+    var err: Boolean = false
+
+    // First attempt at parsing should fail due to attempting to backtrack too far
+    res = dp.parse(input, outputter)
+    err = res.isError()
+    assertTrue(err)
+
+    outputter.reset()
+    try {
+      res = dp.parse(input, outputter)
+    } catch {
+      case e: Abort => {
+        assertTrue(e.getMessage().contains("Usage error"))
+        assertTrue(e.getMessage().contains("invalid input source"))
+      }
+    }
+  }
+
```

Comment:
Would be good to duplicate this in the Java API. Should behave exactlty the same, but in general, anything added as a ScalaAPI test is also added as a Java API test.

Reply:
Done